docs/design/operator.md

```diff
@@ -0,0 +1,414 @@
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+      pluginVolumes:
+      - name: host-run
+        Volumes:
+          hostPath:
+            path: "/run"
+            type: Directory
+        VolumeMounts:
+          name: ''
+          readOnly: true
+          mountPath: "/run"
+          mountPropagation: Bidirectional
+      kubeletDirPath: "/var/lib/kubelet"
+      imagePullPolicy: IfNotPresent
+    provisioner:
+      priorityClassName: system-cluster-critical
+      labels:
+        app: provisioner
+      annotations:
+        k8s.v1.cni.cncf.io/networks: macvlan-conf-1
+      provisionerReplicas: 2
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: storage
+                operator: In
+                values:
+                - node
+      tolerations:
+      - key: storage
+        operator: Exists
+      resources:
+        attacher:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        snapshotter:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        resizer:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        provisioner:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        omapGenerator:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        liveness:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        plugin:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+    liveness:
+      metricsPort: 8000
+    leaderElection:
+      leaseDuration: 100
+      renewDeadline: 100
+      retryPeriod: 10
+    deployCSIAddons: true
+    cephfs:
+      useKernelClient: true
+      kernelMountOptions: ms_mode=secure
+status:
+  phase: Succeeded
+  message: operator config successfully created
+```
+
+### CephCSIDriver CRD
+
+Manages the installation and configuration of CephFS, RBD, and NFS CSI drivers
+within namespaces. Allows customization of driver settings on a per-namespace
+basis. Only single instance of RBD/NFS/CephFS drivers can be created in a
+single namespace, If user wants to deploy new instance of RBD it need to be
+created in a new namespace. The operator need to ensure that no two CR'S are
+created in different namespace that are targetting same csi driver name.
```

Comment:
```
Only single instance of RBD/NFS/CephFS drivers can be created in a
single namespace
```

I do not agree with this restriction, there is no need for it.


```suggestion
Manages the deployment and lifecycle of a single CephFS, RBD, or NFS CSI driver.
Allows customization of driver settings on a per-driver basis. 
```

Reply:
for me it doesn't make sense to allow running two rbd csi drivers in the same namespace. if you don't agree with this restriction we can talk about it.

Reply:
I also prefer the restriction of a single driver instance per namespace. It matches the pattern we have for Rook also that is a single ceph cluster per namespace. It has helped keep some design aspects simpler throughout the project.